src/tool_operate.c

```diff
@@ -1265,7 +1503,17 @@ static CURLcode single_transfer(struct GlobalConfig *global,
         if(result)
           break;
 
-        use_proto = url_proto(per->this_url);
+
+        /* it might have a result from a previous iteration, free proto */
+        Curl_safefree(proto);
+
+        /* result is only used when for ipfs and ipns, ignored otherwise */
+        result = url_proto(&per->this_url, config, &proto);
+        if(result && proto) {
+          if(Tool_safecmp(proto, (char *) "ipfs") ||
+              Tool_safecmp(proto, (char *) "ipns"))
```

Comment:
Since we already check that `proto` is non-NULL three lines up, there is no need for "safe" versions here...